File: day-04/star-1.py
import sys


def parseNbs(string):
    res = []
    curr_nb = 0
    for i in range(len(string)):
        if string[i].isdigit():
            curr_nb *= 10
            curr_nb += int(string[i])
        elif curr_nb > 0:
            res.append(curr_nb)
            curr_nb = 0
    return res


if (len(sys.argv) < 2):
    print("Usage: python", sys.argv[0], "<input file>")
    sys.exit(1)

f = open(sys.argv[1])
lines = f.readlines()

lines = [l.split(':')[1].split('|') for l in lines]

winning_nbs = [parseNbs(l[0]) for l in lines]
my_nbs = [parseNbs(l[1]) for l in lines]

res = 0
for i in range(len(winning_nbs)):
    points = 0
    for nb in winning_nbs[i]:
        if nb in my_nbs[i]:
            points *= 2
            if points == 0:
                points = 1
    res += points

print("Total points:", str(res))

File: day-15/star-1.js
const fs = require('fs');

if (process.argv.length < 3) {
    console.log("Usage: ", process.argv[0], process.argv[1], "<input file>")
    process.exit(1)
}


const hash = (x) => x.reduce((h, c) =>
    ((h + c.charCodeAt(0)) * 17) % 256, 0);



const lines = fs.readFileSync(process.argv[2])
    .toString()
    .replace("\n", "")
    .split(",")
    .map(l => l.split(""));


const hashes = lines.map(l => hash(l));
const res = hashes.reduce((a, b) => a + b, 0);

console.log("result:", res);

